### PR TITLE
KOGITO-5508 Data Index Dev Service

### PR DESCRIPTION
**Pull request by cristianonicolai**
Draft documentation about the Kogito Data Index Dev Service that is coming up in Kogito 1.14 release.

### CODE REVIEW

File: docs/src/main/asciidoc/dev-services.adoc [hunk 1]
```diff
@@ -68,6 +68,12 @@ link:security-openid-connect-dev-services[OIDC Dev Services Guide].
 :no-duration-note: true
 include::{generated-dir}/config/quarkus-keycloak-devservices-keycloak-keycloak-build-time-config.adoc[opts=optional, leveloffset=+1]
 
+== Kogito
+
+The Kogito Dev Service will be enabled when either `kogito-quarkus` or `kogito-quarkus-processes` extension is present in your application. More information can be found at the link:kogito-dev-services[Kogito Dev Services Guide].
+
+//TODO Include config
```
Comment (cristianonicolai) [inline docs/src/main/asciidoc/dev-services.adoc:75]: need some help here on how to include Kogito in the list to automatically generated the extension configuration,

File: docs/src/main/asciidoc/kogito-dev-services-build-time-config.adoc [hunk 2]
```diff
@@ -0,0 +1,54 @@
+[.configuration-legend]
```
Comment (gastaldi) [inline docs/src/main/asciidoc/kogito-dev-services-build-time-config.adoc:1]: Isn't this file auto-generated?

Comment (gsmet) [reply docs/src/main/asciidoc/kogito-dev-services-build-time-config.adoc:1]: It is but in this case it comes from an outside extension.

**Review by gsmet — APPROVED** [review]
Let's get this in, thanks!

**Pull request merged**